twake/frontend/src/app/components/ChatUploads/ChatUploadService.ts

```diff
@@ -0,0 +1,162 @@
+import { FileType, PendingFileType } from 'app/models/File';
+import Api from 'app/services/Api';
+import JWTStorage from 'app/services/JWTStorage';
+import EventEmitter from 'events';
+import RouterServices from 'services/RouterService';
+import Resumable from 'services/uploadManager/resumable';
+import { v1 as uuid } from 'uuid';
+import { isPendingFileStatusPending } from './utils/PendingFiles';
+type ResponseFileType = { resource: FileType };
+
+export enum Events {
+  ON_CHANGE = 'onChange',
+}
+
+export class ChatUploadService extends EventEmitter {
+  private readonly prefixUrl: string = '/internal/services/files/v1';
+  private pendingFiles: PendingFileType[] = [];
+  public currentTaskId: string = '';
+
+  onChange() {
+    this.emit(Events.ON_CHANGE, this.pendingFiles);
+  }
+
+  public async upload(fileList: File[]): Promise<void> {
+    const { companyId } = RouterServices.getStateFromRoute();
+
+    if (!fileList) return;
+
+    if (!this.pendingFiles.some(f => isPendingFileStatusPending(f.state.status))) {
+      //New upload task when all previous task is finished
+      this.currentTaskId = uuid();
+    }
+
+    fileList.forEach(async file => {
+      if (!file) return;
+
+      const pendingFile: PendingFileType = {
+        state: {
+          id: uuid(),
+          file: null,
+          status: 'pending',
+          progress: 0,
+        },
+        uploadTaskId: this.currentTaskId,
+        originalFile: file,
+        resumable: null,
+      };
+
+      this.pendingFiles.push(pendingFile);
+
+      this.onChange();
+
+      // First we create the file object
+      const uploadFileRoute = `${this.prefixUrl}/companies/${companyId}/files?filename=${file.name}&type=${file.type}&total_size=${file.size}`;
+      const resource = (await Api.post<undefined, ResponseFileType>(uploadFileRoute, undefined))
+        ?.resource;
+      if (!resource) {
```

Comment:
Can you create a `FileUploadAPIClient` for this? We should avoid using `Api` directly in such code but rely on API clients ones.

Reply:
Done 👍🏻